### PR TITLE
Add a sync() method to BlockEntityClientSerializable

### PR DESCRIPTION
**Pull request by falkreon**
Currently BlockEntityClientSerializable is missing a clear and side-effect-free way to trigger the actual sync. Marking the block for update on the world itself causes mobs to re-path around the block, so we let the ServerChunkManager know (in the future we might perhaps map `method_14178` as `getServerChunkManager`, as it's a casted version of `getChunkManager`).

This is the sync method working in https://github.com/CottonMC/UnitedConveyors/blob/master/src/main/java/io/github/cottonmc/conveyors/ConveyorBlockEntity.java#L227-L231

### CODE REVIEW

**Comment by falkreon** [comment]
Also, let me know if I should apply this forward to the 1.15 branch, not sure exactly how you prefer me to proceed there.

**Review by ghost — APPROVED** [review]

File: fabric-networking-blockentity-v0/src/main/java/net/fabricmc/fabric/api/block/entity/BlockEntityClientSerializable.java
Comment (Juuxel) [inline fabric-networking-blockentity-v0/src/main/java/net/fabricmc/fabric/api/block/entity/BlockEntityClientSerializable.java]: Unnecessary `public` here (the other methods in the interface don't have them either).

**Review by Juuxel — APPROVED** [review]

**Comment by falkreon** [comment]
Valid. That's part of my code style, I like consistency between abstract methods, interface methods, and method implementations, but here consistency with the rest of the code is key.

File: fabric-networking-blockentity-v0/src/main/java/net/fabricmc/fabric/api/block/entity/BlockEntityClientSerializable.java
Comment (liach) [inline fabric-networking-blockentity-v0/src/main/java/net/fabricmc/fabric/api/block/entity/BlockEntityClientSerializable.java]: Just curious if this will cause some issues when the mod jar is remapped to intermediary names (in mod code the vanilla's remapped method will implement this; however, when it comes to intermediary, that will no longer be the case and the code may crash)

**Comment by grondag** [comment]
> That's part of my code style, I like consistency between abstract methods, interface methods, and method implementations...
 
I feel your pain.

**Comment by grondag** [comment]
> Also, let me know if I should apply this forward to the 1.15 branch, not sure exactly how you prefer me to proceed there.

Can wait until merge time. 1.15 is changing too fast to worry about it now.

**Review by grondag — COMMENTED** [review]
Docs should explain why this is preferable to scheduling block update from world, like the PR header does. 

Fabric format norms include space after cast and newline before blocks when preceded by statements. 

Will usage of this practically always be from within the instance? That seems plausible. 

If not, then perhaps this is better as a static helper. It doesn’t actually require the instance to do its job if position is already known.

**Comment by falkreon** [comment]
I have serious doubts as to whether such a helper method would cause a BlockEntity sync if the BlockEntity did not also implement BlockEntityClientSerializable, so I think usage of this method will safely stay within the instance. And I think separating the interface which supposedly carries the block sync concern, from the method which actually triggers said sync, could cause this sync method to go unnoticed, even with strong javadoc language pointing to it.

**Review by liach — APPROVED** [review]

**Review by grondag — APPROVED** [review]

**Review by Player3324 — COMMENTED** [review]
The feature looks good, I have 2 requests:
- Please rephrased the comment to be like `i am doing xy.\n\n<p>extensive description`. That is having a short "what?" explanation in the 1st paragraph.
- Guarding the cast doesn't appear to make sense with the interface's purposes, I think it should be omitted

**Comment by Player3324** [comment]
Likewise it should likely be an error to call this for a client instance vs just silently ignoring the case.

**Review by liach — APPROVED** [review]

**Review by Player3324 — APPROVED** [review]

**Comment by Player3324** [comment]
Looks good now, thanks!

**Comment by liach** [comment]
Also needs cherry picking to 1.15 a bit later.

**Review by modmuss50 — APPROVED** [review]

File: fabric-networking-blockentity-v0/src/main/java/net/fabricmc/fabric/api/block/entity/BlockEntityClientSerializable.java
Comment (Pyrofab) [inline fabric-networking-blockentity-v0/src/main/java/net/fabricmc/fabric/api/block/entity/BlockEntityClientSerializable.java:27]: Could maybe fix this typo while you are at it (interace)

**Comment by i509VCB** [comment]
I remember that Player mentioned new modules should be ended v1, as example with Dimensions and my Datafixer PR. Is there some level of size you have to reach to make your PR classified as v1 instead of v0?

**Comment by modmuss50** [comment]
v0 is used for modules that were created before we had real modules like we do today. When a major breaking change to a module happens and it needs to be replaced or deperated then a new module can be created with the next number.

**Pull request merged**
